### PR TITLE
Docs: Rename 'Setup' to 'Get started' and 'Start' to 'Run'

### LINKED ISSUE #5949: Create a clear space in docs hierarchy for the task-based docs
Currently, we have the following top-level hierarchy:

image

We don't have a dedicated place to put the task-based documentation. Existing top-level sections seem to overlap.

This issue proposes to have the following top-level sections instead:
- Get started - should include how to quickly install the agent and get hands-on experience. There may be some overlap or links to other pages, but the difference is that "get started" here is dedicated for the new users who are onboarding. We may go less in-depth in these pages and showcase the features and advantages of the Agent.
- Concepts - learn the concepts of the Agent used throughout the documentation. 
- Tasks - pages dedicated to performing specific tasks and achieving a desired outcome. There can be many of them, these should be production-ready and contain best practices and recommendation. They are low level of detail, with Reference pages being the only ones more detailed.
- Tutorials - pages dedicated to learning - usually covering more ground than Tasks, may not need to be production-ready. It's a good place to go to after you've done Get Started section and want to prototype something. After completing a tutorial, the user should be able to dive deeper using Tasks section or Reference. 
- Reference - remains unchanged. The most detailed docs we offer.
- Release notes - unchanged.

Once we have a broad consensus on what sections we want to have, we can assign the existing pages to correct sections.

### PR DESCRIPTION
**Pull request by thampiotr**
#### PR Description

As part of improving docs structure, this PR moves the `flow/setup/` folder to `flow/get-started`. This is following the agreed new guidelines from `writing-docs.md`.

This also renames "Start" to "Run" as the former was clashing with "Get started".

Checked with `make docs` that there are no dead links.
Verified that the aliases for moved pages still works with `curl`.

image

This is the last step 🎉 
Fixes #5949

### CODE REVIEW

File: docs/sources/flow/get-started/start/_index.md
Comment (thampiotr) [inline docs/sources/flow/get-started/start/_index.md:1]: Pages in `start/` folder are new, created in https://github.com/grafana/agent/pull/6087, were not released yet anywhere other than `next` docs preview. So we don't need to add extra backwards compatibility aliases here.

**Comment by rfratto** [comment]
Let me know if I'm nitpicking too hard here, but I find the combination of a folder called "get started" and a subfolder called "start" potentially confusing; people might think that "start" means "start here" (even if it's the second item in the list). 

Should we rename start to another verb to avoid this potential confusion? Or is this too much of a stretch?

(cc @clayton-cornell)

**Comment by thampiotr** [comment]
@rfratto yeah, it could be better. Maybe "Run"? The instructions inside the `start/` folder actually talk about start/stop/restart. Let me know what you think about "Run" - I can make this changes now to reduce the back-n-forth.
Those are still new pages, so should be easier to rename.

**Comment by rfratto** [comment]
run makes sense to me personally 👍

**Comment by thampiotr** [comment]
@rfratto it's done now :) I think we're ready to merge this and I can address any follow-up comments from @clayton-cornell if needed.

**Comment by clayton-cornell** [comment]
Run is good from my side... we might need to tweak the topic titles/headings here and there but it makes sense (and was something I was considering suggesting late yesterday as well... just hadn't got to it)

**Pull request merged**